core/app/models/spree/order.rb

```diff
@@ -686,9 +687,9 @@ def order_total_after_store_credit
 
     def total_applicable_store_credit
       if can_complete? || complete?
-        payments.store_credits.valid.sum(:amount)
+        valid_store_credit_payments.to_a.sum(&:amount)
       else
-        [total, (user.try(:available_store_credit_total, currency: currency) || 0.0)].min
+        [total, (user.try(:available_store_credit_total, currency: currency, force_reload: false) || 0.0)].min
```

Comment:
I see another change of behavior... by the way, this may be totally legit, I did not investigate, can you share your motivation for this change? 🙏